File: buff/classes/354/pub/ch3/aliasing.c
// gcc -Wall -o aliasing aliasing.c && ./aliasing

#include <stdio.h>

int main() {
  union {
    struct { int x,y,z; } s;
    int a[3];
  } u={{0,123,0}};
  u.a[1]++;
  printf("%d %d\n",u.a[1],u.s.y);

  int i=123;
  int *p=&i;
  (*p)++;
  printf("%d %d\n",i,*p);

  int a[100]={0,123};
  extern void f(int a[], int b[]);
  f(a,a);    
  printf("%d\n",a[1]);
}

void f(int a[], int b[]) { a[1]++; b[1]++; }

File: buff/classes/354/pub/ch3/aliasing.f
C     gcc -o aliasing aliasing.f -lgfortran && ./aliasing

      program main
          integer x,y
          equivalence (x,y)
          x=123
          print *,y
          call suba
          call subb
      end

      subroutine suba
          common a,b,c
          a=1
          b=2
          c=3
      end
      
      subroutine subb
          common d,e,f
          print *,d+e+f
      end    
